src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs

```diff
@@ -14,46 +14,100 @@
-        private ServiceClientCredentials _clientCredential;
-        private IAccessToken _accessToken;
+        private string _accessToken = default(string);
         private string _endPoint;
+        private IAzureContext _context;
+        private readonly string _suffix;
+        
         private const string _grantType = "access_token";
-        private const string _scope = "registry:catalog:*";
+        private const string _scopeDefault = "registry:catalog:*";
         private const string _https = "https://";
-        private readonly string _suffix;
+        private const string _refreshTokenKey = "AcrRefreshToken";
+        private const string _acrTokenCacheKey = SharedComponentKeys.AcrTokenCacheKey;
+        private const int _minutesBeforeExpiration = 5;
+
 
         public Action<string> VerboseLogger { get; set; }
         public Action<string> ErrorLogger { get; set; }
         public Action<string> WarningLogger { get; set; }
 
         public ContainerRegistryDataPlaneClient(IAzureContext context)
         {
-            _clientCredential = AzureSession.Instance.AuthenticationFactory.GetServiceClientCredentials(context, AzureEnvironment.Endpoint.ResourceManager);
-            _accessToken = AzureSession.Instance.AuthenticationFactory.Authenticate(context.Account, context.Environment, context.Tenant.Id, null, ShowDialog.Never, null, context.Environment.GetTokenAudience(AzureEnvironment.Endpoint.ResourceManager));
+            _context = context;
             _suffix = context.Environment.ContainerRegistryEndpointSuffix;
-            _client = AzureSession.Instance.ClientFactory.CreateCustomArmClient<AzureContainerRegistryClient>(_clientCredential);
+            ServiceClientCredentials clientCredential = new RenewingTokenCredential(new ExternalAccessToken(_accessToken, () => _accessToken));
+            _client = AzureSession.Instance.ClientFactory.CreateCustomArmClient<AzureContainerRegistryClient>(clientCredential);
+        }
+
+        public IDictionary<string, Tuple<string, DateTime>> TryGetTokenCache()
+        {
+            IDictionary<string, Tuple<string, DateTime>> cache;
+            if (!AzureSession.Instance.TryGetComponent(_acrTokenCacheKey, out cache))
+            {
+                AzureSession.Instance.RegisterComponent<IDictionary<string, Tuple<string, DateTime>>>(_acrTokenCacheKey, () => new Dictionary<string, Tuple<string, DateTime>>());
+                AzureSession.Instance.TryGetComponent(_acrTokenCacheKey, out cache);
+            }
+            return cache;
+        }
+
+        public AzureContainerRegistryClient GetClient()
+        {
+            return _client;
+        }
+
+        public string TryAuthenticate(string scope = _scopeDefault)
+        {
+            _accessToken = TryGetToken(scope);
+            return _accessToken;
+        }
+
+        private string TryGetToken(string key)
+        {
+            IDictionary<string, Tuple<string, DateTime>> cache = TryGetTokenCache();
```

Comment:
Suggest `using AcrTokenCache = IDictionary<string, Tuple<string, DateTime>>` to simplify code